Crash in command completion for `app show`

Seen on branch `main`, commit 14c682e34b0c3c07fc2c2251d2f3cfd3f49dedb8

The command completion for `app show` is not working
An app `hello` was present, and completion did not pick it up.
Trialing the relevant command directly:

```
ep __complete app show h
```

showed a crash.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1292f6a]

goroutine 1 [running]:
golang.org/x/time/rate.(*Limiter).WaitN(0xc00003aa50, 0x0, 0x0, 0x1, 0x0, 0x0)
	/home/work/SUSE/go/pkg/mod/golang.org/x/time@v0.0.0-20201208040808-7e3f01d25324/rate/rate.go:237 +0xba
golang.org/x/time/rate.(*Limiter).Wait(...)
	/home/work/SUSE/go/pkg/mod/golang.org/x/time@v0.0.0-20201208040808-7e3f01d25324/rate/rate.go:219
k8s.io/client-go/util/flowcontrol.(*tokenBucketRateLimiter).Wait(0xc000736ee0, 0x0, 0x0, 0xc0000f2988, 0x0)
	/home/work/SUSE/go/pkg/mod/k8s.io/client-go@v0.20.5/util/flowcontrol/throttle.go:106 +0x4b
k8s.io/client-go/rest.(*Request).tryThrottle(0xc0008f6000, 0x0, 0x0, 0xabcb2c, 0xc0006bd200)
	/home/work/SUSE/go/pkg/mod/k8s.io/client-go@v0.20.5/rest/request.go:587 +0xa4
k8s.io/client-go/rest.(*Request).request(0xc0008f6000, 0x0, 0x0, 0xc0000f2ee8, 0x0, 0x0)
	/home/work/SUSE/go/pkg/mod/k8s.io/client-go@v0.20.5/rest/request.go:860 +0x29d
k8s.io/client-go/rest.(*Request).Do(0xc0008f6000, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/work/SUSE/go/pkg/mod/k8s.io/client-go@v0.20.5/rest/request.go:964 +0xe0
k8s.io/client-go/kubernetes/typed/core/v1.(*nodes).List(0xc00012f0e0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/work/SUSE/go/pkg/mod/k8s.io/client-go@v0.20.5/kubernetes/typed/core/v1/node.go:88 +0x19a
github.com/epinio/epinio/helpers/kubernetes/platform/kind.(*kind).Detect(0xc000466360, 0x0, 0x0, 0xc000458000, 0x0)
	/home/work/SUSE/dev/epinio/helpers/kubernetes/platform/kind/kind.go:25 +0xac
github.com/epinio/epinio/helpers/kubernetes.(*Cluster).detectPlatform(0xc000721c80, 0x0, 0x0)
	/home/work/SUSE/dev/epinio/helpers/kubernetes/cluster.go:127 +0x92
github.com/epinio/epinio/helpers/kubernetes.GetCluster(0x0, 0x0, 0x0, 0x6, 0xc000686ac0)
	/home/work/SUSE/dev/epinio/helpers/kubernetes/cluster.go:106 +0x105
github.com/epinio/epinio/internal/cli/clients.NewEpinioClient(0x0, 0x0, 0xc0000e1900, 0x0, 0x0, 0x0)
	/home/work/SUSE/dev/epinio/internal/cli/clients/client.go:68 +0x60
```

--- a/internal/cli/apps.go
+++ b/internal/cli/apps.go
@@ -1,6 +1,8 @@
 package cli
 
 import (
+	"context"
+
 	v1 "github.com/epinio/epinio/internal/api/v1"
 	"github.com/epinio/epinio/internal/cli/clients"
 	"github.com/pkg/errors"
@@ -90,12 +92,12 @@ var CmdAppCreate = &cobra.Command{
 			return nil, cobra.ShellCompDirectiveNoFileComp
 		}
 
-		app, err := clients.NewEpinioClient(cmd.Context())
+		app, err := clients.NewEpinioClient(context.Background())
 		if err != nil {
 			return nil, cobra.ShellCompDirectiveNoFileComp
 		}
 
-		matches := app.AppsMatching(cmd.Context(), toComplete)
+		matches := app.AppsMatching(context.Background(), toComplete)
 
 		return matches, cobra.ShellCompDirectiveNoFileComp
 	},
@@ -127,12 +129,12 @@ var CmdAppShow = &cobra.Command{
 			return nil, cobra.ShellCompDirectiveNoFileComp
 		}
 
-		app, err := clients.NewEpinioClient(cmd.Context())
+		app, err := clients.NewEpinioClient(context.Background())
 		if err != nil {
 			return nil, cobra.ShellCompDirectiveNoFileComp
 		}
 
-		matches := app.AppsMatching(cmd.Context(), toComplete)
+		matches := app.AppsMatching(context.Background(), toComplete)
 
 		return matches, cobra.ShellCompDirectiveNoFileComp
 	},
@@ -216,12 +218,12 @@ var CmdAppUpdate = &cobra.Command{
 			return nil, cobra.ShellCompDirectiveNoFileComp
 		}
 
-		app, err := clients.NewEpinioClient(cmd.Context())
+		app, err := clients.NewEpinioClient(context.Background())
 		if err != nil {
 			return nil, cobra.ShellCompDirectiveNoFileComp
 		}
 
-		matches := app.AppsMatching(cmd.Context(), toComplete)
+		matches := app.AppsMatching(context.Background(), toComplete)
 
 		return matches, cobra.ShellCompDirectiveNoFileComp
 	},

--- a/internal/cli/delete.go
+++ b/internal/cli/delete.go
@@ -1,6 +1,8 @@
 package cli
 
 import (
+	"context"
+
 	"github.com/epinio/epinio/internal/cli/clients"
 	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
@@ -32,12 +34,12 @@ var CmdDeleteApp = &cobra.Command{
 		if len(args) != 0 {
 			return nil, cobra.ShellCompDirectiveNoFileComp
 		}
-		app, err := clients.NewEpinioClient(cmd.Context())
+		app, err := clients.NewEpinioClient(context.Background())
 		if err != nil {
 			return nil, cobra.ShellCompDirectiveNoFileComp
 		}
 
-		matches := app.AppsMatching(cmd.Context(), toComplete)
+		matches := app.AppsMatching(context.Background(), toComplete)
 
 		return matches, cobra.ShellCompDirectiveNoFileComp
 	},

--- a/internal/cli/env.go
+++ b/internal/cli/env.go
@@ -1,6 +1,8 @@
 package cli
 
 import (
+	"context"
+
 	"github.com/epinio/epinio/internal/cli/clients"
 	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
@@ -51,12 +53,12 @@ var CmdEnvList = &cobra.Command{
 			return nil, cobra.ShellCompDirectiveNoFileComp
 		}
 
-		app, err := clients.NewEpinioClient(cmd.Context())
+		app, err := clients.NewEpinioClient(context.Background())
 		if err != nil {
 			return nil, cobra.ShellCompDirectiveNoFileComp
 		}
 
-		matches := app.AppsMatching(cmd.Context(), toComplete)
+		matches := app.AppsMatching(context.Background(), toComplete)
 
 		return matches, cobra.ShellCompDirectiveNoFileComp
 	},
@@ -91,13 +93,13 @@ var CmdEnvSet = &cobra.Command{
 			return nil, cobra.ShellCompDirectiveNoFileComp
 		}
 
-		app, err := clients.NewEpinioClient(cmd.Context())
+		app, err := clients.NewEpinioClient(context.Background())
 		if err != nil {
 			return nil, cobra.ShellCompDirectiveNoFileComp
 		}
 
 		// #args == 0: application name.
-		matches := app.AppsMatching(cmd.Context(), toComplete)
+		matches := app.AppsMatching(context.Background(), toComplete)
 
 		return matches, cobra.ShellCompDirectiveNoFileComp
 	},
@@ -129,19 +131,19 @@ var CmdEnvShow = &cobra.Command{
 			return nil, cobra.ShellCompDirectiveNoFileComp
 		}
 
-		app, err := clients.NewEpinioClient(cmd.Context())
+		app, err := clients.NewEpinioClient(context.Background())
 		if err != nil {
 			return nil, cobra.ShellCompDirectiveNoFileComp
 		}
 
 		if len(args) == 1 {
 			// #args == 1: environment variable name (in application)
-			matches := app.EnvMatching(cmd.Context(), args[0], toComplete)
+			matches := app.EnvMatching(context.Background(), args[0], toComplete)
 			return matches, cobra.ShellCompDirectiveNoFileComp
 		}
 
 		// #args == 0: application name.
-		matches := app.AppsMatching(cmd.Context(), toComplete)
+		matches := app.AppsMatching(context.Background(), toComplete)
 
 		return matches, cobra.ShellCompDirectiveNoFileComp
 	},
@@ -174,14 +176,14 @@ var CmdEnvUnset = &cobra.Command{
 			return nil, cobra.ShellCompDirectiveNoFileComp
 		}
 
-		// TODO: arg 1: EV match
+		// TODO: match EV names for arg 1 completion
 
-		app, err := clients.NewEpinioClient(cmd.Context())
+		app, err := clients.NewEpinioClient(context.Background())
 		if err != nil {
 			return nil, cobra.ShellCompDirectiveNoFileComp
 		}
 
-		matches := app.AppsMatching(cmd.Context(), toComplete)
+		matches := app.AppsMatching(context.Background(), toComplete)
 
 		return matches, cobra.ShellCompDirectiveNoFileComp
 	},

--- a/internal/cli/service.go
+++ b/internal/cli/service.go
@@ -1,6 +1,7 @@
 package cli
 
 import (
+	"context"
 	"encoding/json"
 
 	"github.com/epinio/epinio/internal/cli/clients"
@@ -46,12 +47,12 @@ var CmdServiceShow = &cobra.Command{
 			return nil, cobra.ShellCompDirectiveNoFileComp
 		}
 
-		app, err := clients.NewEpinioClient(cmd.Context())
+		app, err := clients.NewEpinioClient(context.Background())
 		if err != nil {
 			return nil, cobra.ShellCompDirectiveNoFileComp
 		}
 
-		matches := app.ServiceMatching(cmd.Context(), toComplete)
+		matches := app.ServiceMatching(context.Background(), toComplete)
 
 		return matches, cobra.ShellCompDirectiveNoFileComp
 	},
@@ -79,19 +80,19 @@ var CmdServiceCreate = &cobra.Command{
 			return nil, cobra.ShellCompDirectiveNoFileComp
 		}
 
-		app, err := clients.NewEpinioClient(cmd.Context())
+		app, err := clients.NewEpinioClient(context.Background())
 		if err != nil {
 			return nil, cobra.ShellCompDirectiveNoFileComp
 		}
 
 		if len(args) == 2 {
 			// #args == 2: service plan name.
-			matches := app.ServicePlanMatching(cmd.Context(), args[1], toComplete)
+			matches := app.ServicePlanMatching(context.Background(), args[1], toComplete)
 			return matches, cobra.ShellCompDirectiveNoFileComp
 		}
 
 		// #args == 1: service class name.
-		matches := app.ServiceClassMatching(cmd.Context(), toComplete)
+		matches := app.ServiceClassMatching(context.Background(), toComplete)
 		return matches, cobra.ShellCompDirectiveNoFileComp
 	},
 }
@@ -125,12 +126,12 @@ var CmdServiceDelete = &cobra.Command{
 			return nil, cobra.ShellCompDirectiveNoFileComp
 		}
 
-		app, err := clients.NewEpinioClient(cmd.Context())
+		app, err := clients.NewEpinioClient(context.Background())
 		if err != nil {
 			return nil, cobra.ShellCompDirectiveNoFileComp
 		}
 
-		matches := app.ServiceMatching(cmd.Context(), toComplete)
+		matches := app.ServiceMatching(context.Background(), toComplete)
 
 		return matches, cobra.ShellCompDirectiveNoFileComp
 	},
@@ -148,20 +149,20 @@ var CmdServiceBind = &cobra.Command{
 			return nil, cobra.ShellCompDirectiveNoFileComp
 		}
 
-		app, err := clients.NewEpinioClient(cmd.Context())
+		app, err := clients.NewEpinioClient(context.Background())
 		if err != nil {
 			return nil, cobra.ShellCompDirectiveNoFileComp
 		}
 
 		if len(args) == 1 {
 			// #args == 1: app name.
-			matches := app.AppsMatching(cmd.Context(), toComplete)
+			matches := app.AppsMatching(context.Background(), toComplete)
 			return matches, cobra.ShellCompDirectiveNoFileComp
 		}
 
 		// #args == 0: service name.
 
-		matches := app.ServiceMatching(cmd.Context(), toComplete)
+		matches := app.ServiceMatching(context.Background(), toComplete)
 
 		return matches, cobra.ShellCompDirectiveNoFileComp
 	},
@@ -179,19 +180,19 @@ var CmdServiceUnbind = &cobra.Command{
 			return nil, cobra.ShellCompDirectiveNoFileComp
 		}
 
-		app, err := clients.NewEpinioClient(cmd.Context())
+		app, err := clients.NewEpinioClient(context.Background())
 		if err != nil {
 			return nil, cobra.ShellCompDirectiveNoFileComp
 		}
 
 		if len(args) == 1 {
 			// #args == 1: app name.
-			matches := app.AppsMatching(cmd.Context(), toComplete)
+			matches := app.AppsMatching(context.Background(), toComplete)
 			return matches, cobra.ShellCompDirectiveNoFileComp
 		}
 
 		// #args == 0: service name.
-		matches := app.ServiceMatching(cmd.Context(), toComplete)
+		matches := app.ServiceMatching(context.Background(), toComplete)
 
 		return matches, cobra.ShellCompDirectiveNoFileComp
 	},
@@ -215,12 +216,12 @@ var CmdServiceListPlans = &cobra.Command{
 			return nil, cobra.ShellCompDirectiveNoFileComp
 		}
 
-		app, err := clients.NewEpinioClient(cmd.Context())
+		app, err := clients.NewEpinioClient(context.Background())
 		if err != nil {
 			return nil, cobra.ShellCompDirectiveNoFileComp
 		}
 
-		matches := app.ServiceClassMatching(cmd.Context(), toComplete)
+		matches := app.ServiceClassMatching(context.Background(), toComplete)
 
 		return matches, cobra.ShellCompDirectiveNoFileComp
 	},

--- a/internal/cli/target.go
+++ b/internal/cli/target.go
@@ -1,6 +1,8 @@
 package cli
 
 import (
+	"context"
+
 	"github.com/epinio/epinio/internal/cli/clients"
 	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
@@ -38,7 +40,7 @@ var CmdTarget = &cobra.Command{
 			return nil, cobra.ShellCompDirectiveNoFileComp
 		}
 
-		app, err := clients.NewEpinioClient(cmd.Context())
+		app, err := clients.NewEpinioClient(context.Background())
 		if err != nil {
 			return nil, cobra.ShellCompDirectiveNoFileComp
 		}

